src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs

```diff
@@ -234,9 +240,16 @@ private void UpdateRestorePlan(RestoreDatabaseTaskDataObject restoreDataObject)
                 restoreDataObject.AddFile(restoreDataObject.RestoreParams.BackupFilePath);
             }
             restoreDataObject.RestorePlanner.ReadHeaderFromMedia = !string.IsNullOrEmpty(restoreDataObject.RestoreParams.BackupFilePath);
-            var dbNames = restoreDataObject.GetSourceDbNames();
-            string dbName = dbNames.First();
-            restoreDataObject.RestorePlanner.DatabaseName = dbName;
+           
+            restoreDataObject.RestorePlanner.DatabaseName = restoreDataObject.DefaultDbName;
+            restoreDataObject.TargetDatabase = restoreDataObject.RestoreParams.DatabaseName;
+            //TODO: used for other types of restore
+            /*bool isTailLogBackupPossible = restoreDataObject.RestorePlanner.IsTailLogBackupPossible(restoreDataObject.RestorePlanner.DatabaseName);
```

Comment:
yes it's not used for full backup

---
In reply to: [127762843](https://github.com/Microsoft/sqltoolsservice/pull/414#discussion_r127762843) [](ancestors = 127762843)